### ISSUE TITLE
How to create a map step by step?

### ISSUE THREAD

**Issue by Derekabc** [carla-simulator/carla#1572]
Hello, I am new to Carla. Now I want to create a new map by RoadRunner and Unreal Engine 4. 
I really appreciate there are so many contributors to this topic. I mainly focus on these two tutorials for building my own map. 
Guide#1: [https://github.com/carla-simulator/carla/issues/1015](url) and 
Guide#2: [https://carla.readthedocs.io/en/latest/how_to_make_a_new_map](url)
I find it's pretty hard to follow these instructions to build my own map. One reason is that I do not know how to check each step is correct or not so it also takes me a lot of time to figure out this. There are my steps and questions:
1. created a map on RoadRunner. Firstly, I changed the Rotation to 180 degree followed by guide#2. Then drew my map, just a simple one. **I do not know what is OpenDRIVE Preview Tool, does this step help?** I simply created my map and exported as fbx and opendrive files. Saved using same name, test1.fbx and test1.xodr.   
2. I copied the opendrive file to `/home/dong/carla/Unreal/CarlaUE4/Content/Carla/Maps/OpenDrive/test1.xodr`. 
3. open the unreal engine and create a new level and saved it to `Maps/test`. Then imported the fbx file to a newly created folder but I got some warinings. Changed the roation to (0, 0, 0).
![Screenshot from 2019-04-25 15-11-06](https://user-images.githubusercontent.com/25771207/56774304-1104f580-6790-11e9-9b5a-6efda1c716cf.png)
I got the RoadRunner plugins installed. I do not know why I cannot see the results just like this video shows. [https://www.youtube.com/watch?v=cu_IwFZhvhI](url), I need to drag the meshes to the levele by myself.  
![Screenshot from 2019-04-25 15-21-31](https://user-images.githubusercontent.com/25771207/56774330-2e39c400-6790-11e9-815c-eaccf4644a95.png)
3. I just continued to create my own map.  I dragged all the items to the level and deleted the floor in the newly created level. Then I searched the OpenDrive actor / CarlaMapGenerator and dragged them to the level.  Check "generate routes"  and then get 
![Screenshot from 2019-04-25 15-42-05](https://user-images.githubusercontent.com/25771207/56775559-8e7f3480-6795-11e9-8edb-16c99bb29b28.png)
I found it's difficult to move the route to the road I generated. Later I tried another created road by the same way, I found I cannot see the route anymore after clicking "generate routes" . 
Also a log information `LogCarla: Loading OpenDrive file '../../../../carla/Unreal/CarlaUE4/Content/Carla/Maps/OpenDrive/test1.xodr'
WARNING: cannot parse georeference: ''. Using default values. `
I want to know which step may I make mistakes? 
![Screenshot from 2019-04-25 20-15-19](https://user-images.githubusercontent.com/25771207/56775829-e4081100-6796-11e9-887a-9a8afbdab4b7.png)
Thank you for your help!!

**Comment by Derekabc** [comment]
@XGodina hi, I still need your help. thanks

**Comment by eloy96vg** [comment]
@Derekabc i have used generate_map.py as they explain on this page (https://github.com/carla-simulator/carla/blob/master/Docs/generate_map_from_fbx.md) and have done the whole process without doing anything. It is easier than doing the manual process.

**Comment by Derekabc** [comment]
@eloy96vg Thank you for your reply !  In this method, can we add new spawn points and path?

**Comment by aisichongzi007** [comment]
hi , how to get  the  key  of  roadrunner.   it,s  expensive.
or ,do you have the crack of roadrunner

**Comment by hieunguyen88** [comment]
Hi @Derekabc , how do you generate the route manually? I drag OpenDriveActor and CarlaMapGenerator, but I didn't see any route show up. You mentioned "clicking 'generate routes'", is there a button in UnrealEditor? I couldn't find it.

About your issue, I think the map need at least one junction (I run into the same issue when my map only has one straight road. It works when there is at least one junction). Try to make one junction in road runner and import it again

**Comment by athreyasridharan** [comment]
@hieunguyen88 @Derekabc I'm also looking for the generate routes button. Could you point me to it?

**Comment by Derekabc** [comment]
hi, @athreyasridharan, you see the last picture of my first comment. You can find the generate route button when you choose the CarlaMapGenerate label.

**Comment by RehabHsn** [comment]
@eloy96vg  the link is not working 
could you please update the link

**Comment by mehmor** [comment]
> Hello, I am new to Carla. Now I want to create a new map by RoadRunner and Unreal Engine 4.
> I really appreciate there are so many contributors to this topic. I mainly focus on these two tutorials for building my own map.
> Guide#1: [https://github.com/carla-simulator/carla/issues/1015](url) and
> Guide#2: [https://carla.readthedocs.io/en/latest/how_to_make_a_new_map](url)
> I find it's pretty hard to follow these instructions to build my own map. One reason is that I do not know how to check each step is correct or not so it also takes me a lot of time to figure out this. There are my steps and questions:
> 
> 1. created a map on RoadRunner. Firstly, I changed the Rotation to 180 degree followed by guide#2. Then drew my map, just a simple one. **I do not know what is OpenDRIVE Preview Tool, does this step help?** I simply created my map and exported as fbx and opendrive files. Saved using same name, test1.fbx and test1.xodr.
> 2. I copied the opendrive file to `/home/dong/carla/Unreal/CarlaUE4/Content/Carla/Maps/OpenDrive/test1.xodr`.
> 3. open the unreal engine and create a new level and saved it to `Maps/test`. Then imported the fbx file to a newly created folder but I got some warinings. Changed the roation to (0, 0, 0).
>    ![Screenshot from 2019-04-25 15-11-06](https://user-images.githubusercontent.com/25771207/56774304-1104f580-6790-11e9-9b5a-6efda1c716cf.png)
>    I got the RoadRunner plugins installed. I do not know why I cannot see the results just like this video shows. [https://www.youtube.com/watch?v=cu_IwFZhvhI](url), I need to drag the meshes to the levele by myself.
>    ![Screenshot from 2019-04-25 15-21-31](https://user-images.githubusercontent.com/25771207/56774330-2e39c400-6790-11e9-815c-eaccf4644a95.png)
> 4. I just continued to create my own map.  I dragged all the items to the level and deleted the floor in the newly created level. Then I searched the OpenDrive actor / CarlaMapGenerator and dragged them to the level.  Check "generate routes"  and then get
>    ![Screenshot from 2019-04-25 15-42-05](https://user-images.githubusercontent.com/25771207/56775559-8e7f3480-6795-11e9-8edb-16c99bb29b28.png)
>    I found it's difficult to move the route to the road I generated. Later I tried another created road by the same way, I found I cannot see the route anymore after clicking "generate routes" .
>    Also a log information `LogCarla: Loading OpenDrive file '../../../../carla/Unreal/CarlaUE4/Content/Carla/Maps/OpenDrive/test1.xodr' WARNING: cannot parse georeference: ''. Using default values. `
>    I want to know which step may I make mistakes?
>    ![Screenshot from 2019-04-25 20-15-19](https://user-images.githubusercontent.com/25771207/56775829-e4081100-6796-11e9-887a-9a8afbdab4b7.png)
>    Thank you for your help!!

What if we don't have Roadrunner? what is the best option?

**Comment by ndeshp2s** [comment]
Hello,

I followed the guide mentioned here [here](https://carla.readthedocs.io/en/latest/how_to_make_a_new_map/)
to create my custom environment map. I did not get any error, however, after importing the .fbx file into unreal engine the objects in the environment have no textures. 

Why are not the material textures/colors visible in unreal? 

Road Runner map:
![Screenshot from 2019-08-21 19-00-59](https://user-images.githubusercontent.com/1629735/63453089-27724380-c448-11e9-89e6-f5d7460e7bfa.png)

Unreal engine map:
![Screenshot from 2019-08-21 19-09-36](https://user-images.githubusercontent.com/1629735/63453107-3527c900-c448-11e9-9bf5-7c6cb53d35f3.png)
![Screenshot from 2019-08-21 19-04-31](https://user-images.githubusercontent.com/1629735/63453110-378a2300-c448-11e9-9475-db453f163824.png)

**Comment by Kingsman1995** [comment]
Hello, i want to ask where did u get the RoadRunner install file? Because i've contacted the Vectorzero and gotten no response yet.

**Comment by GeraldGuo7** [comment]
It looks like Carla has some connection with Mathworks and does anyone find a way to use other software to generate .fbx and .xodr files?
If I just want to use a simple map which does not have road standards, can I just export the .fbx file to do the simulation?
Thanks in advance.
